### DISCUSSION TITLE
Proposal: Null-propagating operator on property values

### DISCUSSION THREAD

**Discussion by Toemsel** [dotnet/csharplang#1258] [General]
With C# 6 came the null-propagating operator feature. Such as

`randomObject?.DoStuff();`

What I always stumble up on is following _issue_

`someObject?.Property = val;`

Which leads to "CS0131	The left-hand side of an assignment must be a variable, property or indexer	DataAccess"

It would be great to set property values with the null-propagating operator if the object is not null. Under the hood something like:

`if(someObject != null)
    someObject.Property = val;`

**Comment by orthoxerox** [comment]
Why not fields and indexers too?

**Comment by MkazemAkhgary** [comment]
#1106

**Comment by jnm2** [comment]
Yes, please close so that discussion moves to #1106.
